saleor/checkout/utils.py

```diff
@@ -912,6 +960,19 @@ def _process_user_data_for_order(checkout):
     }
 
 
+def validate_gift_cards(checkout: Checkout):
+    """Check if all gift cards assigned to checkout are available."""
+    if (
+        not checkout.gift_cards.count()
+        == checkout.gift_cards.active(date=date.today()).count()
+    ):
+        msg = pgettext(
+            "Gift card not applicable",
+            "Gift card expired in meantime. Order placement aborted.",
```

Comment:
Check with @timcla if this message is correct.

Reply:
'Gift card not applicable' seems clear. i.e. these products cannot be purchased with that gift card

'Gift card expired in meantime. Order placement aborted' could be changed to:
'Gift card has expired. Order placement cancelled'